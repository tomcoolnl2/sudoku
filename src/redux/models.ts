
import { GridMatrix, GridMatrixCoörds, GridMatrixSeries, SudokuInputValue } from '../typings'


export interface AppState {
    initialGameMatrix: GridMatrix // initial grid only showing clue numbers
    solutionMatrix: GridMatrix // for validating user input
    workingMatrix: GridMatrix // the user will use this grid
    mistakesMatrix: GridMatrix // mistakes will be tracked
    trackedInput: GridMatrixSeries
    trackedMistakes: number
    selection: GridMatrixCoörds
    selectedInputValue: SudokuInputValue
    settings: AppSettings
}

export interface AppSettings {
    highlightDuplicates: boolean
    allowedMistakes: number
    darkModeEnabled: boolean
}