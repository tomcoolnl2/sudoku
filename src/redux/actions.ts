
import { Action, AnyAction } from 'redux'
import { GridMatrixCoörds, SudokuInputValue } from '../typings'
import * as types from './types'


export const createGrid = (): Action => ({ type: types.UNLEASH_THE_MATRIX })

export const fillCell = (value: SudokuInputValue, coords: GridMatrixCoörds): AnyAction => ({ 
	value,
	coords,
	type: types.FILL_CELL
})

export const selectCell = (coords: GridMatrixCoörds): AnyAction => ({ 
	coords,
	type: types.SELECT_CELL
})
