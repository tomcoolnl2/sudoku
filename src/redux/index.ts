
import { reducer } from './reducer'
export type StoreReducer = ReturnType<typeof reducer>
export { reducer }

export { configureStore } from './configureStore'

export * from './actions'
export * from './models'