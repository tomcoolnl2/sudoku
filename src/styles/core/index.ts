
import { createGlobalStyle, css } from 'styled-components'


export const theme = {
	colors: {
		background: 'radial-gradient(#282c34cc, #282c34)',
		black: '#282c34',
		blue: '#a0e9fd',
		lightBlue: '#caf3fe',
		white: 'white'
	},
	transition: '0.3s'
}

export const GlobalStyles = createGlobalStyle`
    ${props => css`
        html {
            height: 100%;
        }

        body {
            display: flex;
            flex-direction: column;
            height: 100%;
            margin: 0;

            #root {
                background-image: ${props.theme.colors.background};
                color: ${props.theme.colors.black};
                display: flex;
                font-family: sans-serif;
                height: 100%;
                justify-content: center;
                padding: 15px;
            }
        }
    `
}`