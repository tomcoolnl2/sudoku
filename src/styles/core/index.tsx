
import { createGlobalStyle, css } from 'styled-components'

const theme = {
	colors: {
		background: 'radial-gradient(#282c34cc, #282c34)',
		primary: 'white',
		secondary: '#282c34',
		active: '#EDDA74',
		highlighted: '#FCF3CF',
		errorPrimary: '#CB4335',
		errorSecundary: '#E6B0AA'
	},
	transition: '0.3s'
}

export const lightTheme = {
	...theme
}

export const darkTheme = {
	...theme,
	colors: {
		...theme.colors,
		primary: '#2E4053',
		secondary: '#EBEDEF',
		active: '#515A5A',
		highlighted: '#616A6B',
	},
}

export const GlobalStyles = createGlobalStyle`
    ${props => css`
		:root {
			color-scheme: light dark;
		}

        html {
            height: 100%;
        }

        body {
            display: flex;
            flex-direction: column;
            height: 100%;
            margin: 0;

            #root {
                background-image: ${props.theme.colors.background};
                color: ${props.theme.colors.secondary};
                display: flex;
                font-family: sans-serif;
                height: 100%;
                justify-content: center;
                padding: 15px;
            }
        }
    `
}`