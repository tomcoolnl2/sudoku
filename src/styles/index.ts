
export * from './title'
export * from './content'
export * from './title'
export * from './card'
export * from './grid'
export * from './block'