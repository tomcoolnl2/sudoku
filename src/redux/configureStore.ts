
import { createStore } from 'redux'
import { devToolsEnhancer } from 'redux-devtools-extension'
import { persistStore, persistReducer} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import { reducer } from '.'


const persistConfig = {
	key: 'sudoku',
	storage
}

const persistedReducer = persistReducer(persistConfig, reducer)

export function configureStore(): unknown {
    
	const store = createStore(
		persistedReducer,
		devToolsEnhancer({})
	)

	const persistor = persistStore(store)

	return { store, persistor }
}