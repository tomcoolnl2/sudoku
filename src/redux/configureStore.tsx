
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { devToolsEnhancer } from 'redux-devtools-extension'
import { persistStore, persistReducer} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import { reducer } from '.'
import { FC } from 'react'


const persistConfig = {
	key: 'sudoku',
	storage
}

const persistedReducer = persistReducer(persistConfig, reducer)

export function configureStore(): unknown {
    
	const store = createStore(
		persistedReducer,
		devToolsEnhancer({})
	)

	const persistor = persistStore(store)

	return { store, persistor }
}

/**
 * Wrapper for unit/snapshot testing
 * @param Component React children
 * @returns void
 */
export const withRedux: FC = (Component: unknown) => {
	// Weird validation error causes the app to crash, but it works fine
	// eslint-disable-next-line
	// @ts-ignore
	const { store } = configureStore()
	return (
		<Provider store={store}>
			{Component}
		</Provider>
	)
}