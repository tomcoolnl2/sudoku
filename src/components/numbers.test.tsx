
import { render } from '@testing-library/react'
import { withRedux } from '../redux/configureStore'
import { withTheme } from '../styles/core'
import { Numbers } from './'


describe('Numbers', () => {
	
	it('should match it\'s snapshot', () => {
		const { container } = render(withTheme(withRedux(<Numbers />)))
		expect(container.firstChild).toMatchSnapshot()
	})
})