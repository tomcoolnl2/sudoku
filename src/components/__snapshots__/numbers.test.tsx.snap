// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numbers should match it's snapshot 1`] = `
<div
  class="sc-iBPTik kafkLt"
>
  <button
    class="sc-gKseQn cjqTfh"
  >
    1
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    2
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    3
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    4
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    5
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    6
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    7
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    8
  </button>
  <button
    class="sc-gKseQn cjqTfh"
  >
    9
  </button>
</div>
`;
